strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/proxied/ProxyRepositoryArtifactResolver.java

```diff
@@ -0,0 +1,152 @@
+ */
+public abstract class ProxyRepositoryArtifactResolver
+{
+
+    @Inject
+    protected ConfigurationManager configurationManager;
+
+    @Inject
+    protected RemoteRepositoryAlivenessCacheManager remoteRepositoryAlivenessCacheManager;
+
+    @Inject
+    protected LayoutProviderRegistry layoutProviderRegistry;
+
+    @Inject
+    protected ArtifactEventListenerRegistry artifactEventListenerRegistry;
+
+    @Inject
+    protected ProxyRepositoryConnectionPoolConfigurationService proxyRepositoryConnectionPoolConfigurationService;
+
+    public InputStream getInputStream(final String storageId,
+                                      final String repositoryId,
+                                      final String path)
+            throws IOException, ArtifactTransportException, NoSuchAlgorithmException, ProviderImplementationException
+    {
+        final Storage storage = getConfiguration().getStorage(storageId);
+        final Repository repository = storage.getRepository(repositoryId);
+
+        getLogger().debug("Checking in " + storage.getId() + ":" + repositoryId + "...");
+
+        final InputStream candidate = preProxyRepositoryAccessAttempt(repository, path);
+        if (candidate != null)
+        {
+            return candidate;
+        }
+
+        final RemoteRepository remoteRepository = repository.getRemoteRepository();
+        if (!remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository))
+        {
+            getLogger().debug("Remote repository '" + remoteRepository.getUrl() + "' is down.");
+
+            return null;
+        }
+
+        final ArtifactResolver client = new ArtifactResolver(proxyRepositoryConnectionPoolConfigurationService.getRestClient());
+        client.setRepositoryBaseUrl(remoteRepository.getUrl());
+        client.setUsername(remoteRepository.getUsername());
+        client.setPassword(remoteRepository.getPassword());
+
+        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        final String resourcePath = layoutProvider.resolveResourcePath(repository, path);
+
+        try (final CloseableProxyRepositoryResponse closeableProxyRepositoryResponse =
+                     new CloseableProxyRepositoryResponse(client.getResourceWithResponse(resourcePath)))
+        {
+            final Response response = closeableProxyRepositoryResponse.response;
+
+            if (response.getStatus() != 200 || response.getEntity() == null)
+            {
+                return null;
+            }
+
+            InputStream is = response.readEntity(InputStream.class);
+            if (is == null)
+            {
+                return null;
+            }
+
+            is = onSuccessfulProxyRepositoryResponse(is, storageId, repositoryId, path);
+
+            final RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
+            if (!layoutProvider.isChecksum(artifactPath) && !layoutProvider.isMetadata(path))
```

Comment:
@fuss86 I think that it's better to use `RepositoryFileAttributes` here like follows:
```
    RepositoryFileAttributes a = (RepositoryFileAttributes) Files.readAttributes(path, BasicFileAttributes.class);
    if (!a.isChecksum() && !a.isMetadata())
    {
        ...
    }
```

Reply:
@carlspring, should I proceed with this change in some another pull request ?

Reply:
Yes, please.